.github/workflows/release.yaml

```diff
@@ -0,0 +1,90 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    outputs:
+      published: ${{ steps.release.outputs.published }}
+      publishedPackages: ${{ steps.release.outputs.publishedPackages }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.node-version'
+          cache: 'yarn'
+          cache-dependency-path: "yarn.lock"
+
+      - name: Install Dependencies
+        run: yarn
+
+      - name: Create Release Pull Request or Tag
+        id: changesets
+        uses: changesets/action@v1
+        with:
+          version: yarn run version
+          publish: yarn run release
+          commit: 'ci(changesets): version packages'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Release output
+        id: release
+        run: |
+          # intermediate variable necessary to avoid quotes around the array
+          # that prevented the fromJson function to work correctly
+          publishedPackages=${{toJSON(steps.changesets.outputs.publishedPackages)}}
+
+          echo 'published=${{steps.changesets.outputs.published}}' >> "$GITHUB_OUTPUT"
+          echo "publishedPackages=$publishedPackages" >> "$GITHUB_OUTPUT"
+
+  publish:
+    name: Publish
+    needs: [release]
+    if: needs.release.outputs.published == 'true'
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      matrix:
+        package: ${{ fromJson(needs.release.outputs.publishedPackages) }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.node-version'
+          cache: 'yarn'
+          cache-dependency-path: "yarn.lock"
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Prune workspace
+        run: yarn turbo prune ${{ matrix.package.name }} --out ./bundle
+
+      - name: Create bundle
+        run: zip -r ${{ matrix.package.name }}-${{ matrix.package.version }}.zip ./bundle
```

Comment:
what is the purpose of this `bundle` ?

Reply:
It's the file attached to the release. The release will then contain a file bundle ready to be deployed/published (without the need to re-init the environment and perform other builds)

Reply:
but this `bundle.zip` is just a pruned monorepo, it is not directly deployable

Reply:
Ok, delete my previous comment I was probably drunk sorry 🙃 

One point still applies: the release contains a file archive with the code to run the application/package. We then have to `yarn && yarn build`. The idea (although incomplete) is that you can publish a release by just using its attached bundles.

Reply:
I thought we wanted to review the that part in next steps, but it's ok to use it as is

Reply:
then  I suggest you not attach any zip to the release for the moment. We'll do this later when we're sure what we want to do